### PR TITLE
Enable has fixes for some specific cases

### PR DESCRIPTION
**Pull request by lukechurch**
@devoncarew PTAL

### CODE REVIEW

File: lib/src/analyzer.dart [hunk 1]
```diff
@@ -263,6 +265,17 @@ class _Logger extends engine.Logger {
 }
 
 class _Error {
+  final _HAS_FIXES_WHITELIST = [
```
Comment (devoncarew) [inline lib/src/analyzer.dart]: `static`?

File: lib/src/analyzer.dart [hunk 2]
```diff
@@ -263,6 +265,17 @@ class _Logger extends engine.Logger {
...
+    HintCode.DEAD_CODE,
+    HintCode.DUPLICATE_IMPORT,
+    HintCode.UNUSED_IMPORT,
+    HintCode.UNUSED_LOCAL_VARIABLE,
+    ParserErrorCode.EXPECTED_TOKEN,
+    StaticWarningCode.UNDEFINED_CLASS,
+    StaticWarningCode.UNDEFINED_CLASS_BOOLEAN,
+    ];
```
Comment (devoncarew) [inline lib/src/analyzer.dart]: nit: unindent 2 spaces

File: test/analyzer_test.dart [hunk 3]
```diff
@@ -50,6 +50,34 @@ void defineTests() {
         expect(results.issues.length, 3);
       });
     });
+
+    test('missing ;', () {
+      return analyzer.analyze(r'''
+void main() {
```
Comment (devoncarew) [inline test/analyzer_test.dart]: This reads odd - can you just make this:
```
'void main() {\n  int i = 55\n}'
```

Comment (devoncarew) [inline test/analyzer_test.dart]: or hoist it to a local var

File: test/analyzer_test.dart [hunk 4]
```diff
@@ -50,6 +50,34 @@ void defineTests() {
...
+
+             results.issues.where(
+               (issue) => issue.message == "Expected to find ';'")
+               .forEach((issue) {
+                 _missingSemiC++;
+                 expect(issue.hasFixes, true);
+             });
+             expect(_missingSemiC, 1);
+
+           });
```
Comment (devoncarew) [inline test/analyzer_test.dart]: unindent this block 4 spaces

File: test/analyzer_test.dart [hunk 5]
```diff
@@ -50,6 +50,34 @@ void defineTests() {
...
+    });
+
+    test('no fixes', () {
+      return analyzer.analyze(r'''#''')
+          .then((AnalysisResults results) {
+             expect(results.issues.length, 2);
+             results.issues.forEach((issue) => expect(issue.hasFixes, false));
+             });
```
Comment (devoncarew) [inline test/analyzer_test.dart]: same here, back it up 4 spaces

**Comment by devoncarew** [comment]
lgtm; some ws nits

File: lib/src/analyzer.dart (hunk 1 above)
Comment (lukechurch) [inline lib/src/analyzer.dart]: yes! done.

File: lib/src/analyzer.dart (hunk 2 above)
Comment (lukechurch) [inline lib/src/analyzer.dart]: done

File: test/analyzer_test.dart (hunk 4 above)
Comment (lukechurch) [inline test/analyzer_test.dart]: done

File: test/analyzer_test.dart (hunk 5 above)
Comment (lukechurch) [inline test/analyzer_test.dart]: done.

**Comment by lukechurch** [comment]
Thanks @devoncarew and address a suggestion from @kasperpeulen

**Pull request merged**